includes/Checker/Preparations/Use_Minimal_Theme_Preparation.php

```diff
@@ -125,10 +125,14 @@ public function get_theme_slug() {
 	 *
 	 * @since n.e.x.t
 	 *
+	 * @param string $current Current theme.
 	 * @return string The theme name.
 	 */
-	public function get_theme_name() {
+	public function get_theme_name( $current ) {
```

Comment:
What is the purpose of this new parameter? That seems pointless to me as you're just passing an empty string to it, I think it should be removed.

Reply:
This is a callback to a filter. So if the theme doesn't exist, then it should fallback to the current value.

Reply:
I agree with @spacedmonkey here, we're using this method for the `pre_option_current_theme` filter and this update falls back to the current theme name if our minimal theme doesn't exist.

The empty string being passed is used in tests where we're expecting the minimal theme name. It may be worth adding another test for when the `Use_Minimal_Theme_Preparation` class uses a theme that does not exist to ensure the fallback is returned.

Reply:
That's a fair point on this being a filter callback. However, I don't think we should silently use the regular theme if the theme does not exist.

This class is to _enforce_ the use of our specific minimal test theme: If the test theme for whatever reason does not exist, that means something is wrong with our code. Falling back to the default theme is not the right thing to do as that will result in incorrect runtime data that is affected by the actual theme used on the site.

I think we should throw an exception if the theme does not exist and then get rid of the `$current` parameter again as for this particular filter callback it should never be used.

Reply:
I guess it depends on how we want to handle this error state. 
1. Default to current theme - do nothing. 
2. Call `wp_die` with error message. 
3. Call `_doing_it_wrong`  with error message and default to current theme. 
4. Throw an exception and trigger fatal error handling. 

This is super edge, so I don't think we should make a big deal of this. I will defer to others thoughts on this.

Reply:
Because it's indeed super edge case, I think we should just throw an exception and be done with it.

Reply:
Thanks @felixarntz I agree with throwing an exception here 👍

Reply:
Now throws `exception` .